controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -197,7 +197,8 @@ protected void startUp() {
                 // Create ControllerEventProcessor object.
                 controllerEventProcessors = new ControllerEventProcessors(host.getHostId(),
                         serviceConfig.getEventProcessorConfig().get(), localController, checkpointStore, streamStore,
-                        hostStore, segmentHelper, connectionFactory, streamMetadataTasks, controllerExecutor);
+                        hostStore, segmentHelper, connectionFactory, streamMetadataTasks, streamTransactionMetadataTasks,
```

Comment:
Unrelated to this PR, but I was wondering if we should refactor the constructor of `ControllerEVentProcessors` to take something like a collection so that we don't have to add a new parameter every time we come up with a new task. Otherwise, the list of parameters will grow indefinitely, although granted that I don't know how many more of those we will and want to have.

Reply:
Actually we have helper modules and stores that we pass here because those modules help us implement our workflows in event processors.
This is a fairly contained list and wont suddenly grow large. There may be one or two helper modules or new stores we may add in future depending on the requirement that comes up in future. But this list wont grow suddenly as we have covered all our top level stores and helper modules here. 

Here is a list of modules and stores we have in our system as of now: 
1. segment helper (to interact with segment store)
2. stream metadata store (stream metadata)
3. host store (cluster management, segment container ownership)
4. stream metadata tasks -> business logic to start stream specific tasks {historically all tasks were implemented here as part of task framework. Then we moved things to event processor and this became a helper module}. 
5. stream transaction metadata tasks -> business logic to create transaction and start transaction commit and abort.
6. checkpoint store for storing event processor checkpoints.